### PR TITLE
Drop Python 2 and Python 3.4 support

### PR DESCRIPTION
**Pull request by mlech-reef**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -39,6 +33,8 @@ b2sdk>=0.0.0,<1.0.0
 
 ## Not released yet
 
+* Drop Python 2 and Python 3.4 support :tada:
```
Comment (ppolewicz) [inline README.md:36]: please add a link to https://www.backblaze.com/b2/docs/quick_command_line.html

File: b2sdk/raw_simulator.py [hunk 2]
```diff
@@ -1038,8 +1033,7 @@ def list_buckets(
 
         # Do the query
         sorted_buckets = [
-            self.bucket_name_to_bucket[name]
-            for name in sorted(six.iterkeys(self.bucket_name_to_bucket))
+            self.bucket_name_to_bucket[name] for name in sorted(self.bucket_name_to_bucket)
```
Comment (ppolewicz) [inline b2sdk/raw_simulator.py:1036]: wouldn't sorting output of `.items()` be faster here due to not needing to first go through N to get every key and then retrieving every value with the theoretical "O(1)" which really is O(log N)? I mean, there's a max of 100 entries in this thing anyway so maybe it doesn't matter much, but I'm wondering why you chose to write it this way

File: b2sdk/sync/folder.py [hunk 3]
```diff
@@ -282,12 +280,8 @@ def _handle_non_unicode_file_name(cls, name):
         names.
         """
         # if it's all ascii, allow it
-        if six.PY2:
-            if all(ord(c) <= 127 for c in name):
-                return name
-        else:
-            if all(b <= 127 for b in name):
-                return name
+        if all(b <= 127 for b in name):
```
Comment (ppolewicz) [inline b2sdk/sync/folder.py:283]: by the way, isn't there a faster way to do this using `str.encode('ascii')` and catching an exception?

File: b2sdk/raw_simulator.py (hunk 2 above)
Comment (mlech-reef) [reply b2sdk/raw_simulator.py:1036]: I didn't choose to write it this way :) For most parts, I used regexes and structural replacement to get rid of six. Then I did a review for myself without analyzing the code for quality and speed, only for correctness.

I will update it

Comment (mlech-reef) [reply b2sdk/raw_simulator.py:1036]: It did what six is doing under the hood for Python 3 with few exceptions, so it didn't introduce any regression.

File: b2sdk/sync/folder.py (hunk 3 above)
Comment (mlech-reef) [reply b2sdk/sync/folder.py:283]: Trying to decode bytes is a lot slower. This method should not be called if `self.root` is `str`. In my opinion, we should prevent using bytes for paths and raise an exception in such case

File: b2sdk/raw_simulator.py (hunk 2 above)
Comment (mlech-reef) [reply b2sdk/raw_simulator.py:1036]: Using `items()` is actually 2x slower because of tuple lookups and unpacking

Comment (ppolewicz) [reply b2sdk/raw_simulator.py:1036]: 🤦

File: b2sdk/sync/folder.py (hunk 3 above)
Comment (ppolewicz) [reply b2sdk/sync/folder.py:283]: yeah I think we should move it all to pathlib in v2 and leave wrappers for strings for v1. I updated the ticket with the list of things to consider for apiver v2.

File: README.md (hunk 1 above)
Comment (ppolewicz) [reply README.md]: actually, I was wrong, that link is for b2cli, not for b2sdk!

Comment (mlech-reef) [reply README.md:36]: I have removed the last commit

**Pull request merged**